From 98bb836c3c73632ce00f7404e0a1c0957a72f834 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 29 Aug 2018 10:04:57 +0800
Subject: [PATCH 4473/5242] MLK-19328-2: hdp: add channel map for hdmi audio

commit  a8ab2fab9ba40e8f4d75af78c0f446ce8533e2b7 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add channel map for hdmi audio, originally it is in
audio info frame function, but removed by
commit 6b97462b6407 ("MLK-18690-3: hdp api: Remove
info frame API function")

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp-audio.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp-audio.c b/drivers/gpu/drm/imx/hdp/imx-hdp-audio.c
index dace3cc..0a509a1 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp-audio.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp-audio.c
@@ -72,6 +72,13 @@ static void imx_hdmi_audio_avi_set(state_struct *state,
 	frame.channels = channels;
 	frame.coding_type = HDMI_AUDIO_CODING_TYPE_STREAM;
 
+	if (channels == 2)
+		frame.channel_allocation = 0;
+	else if (channels == 4)
+		frame.channel_allocation = 0x3;
+	else if (channels == 8)
+		frame.channel_allocation = 0x13;
+
 	ret = hdmi_audio_infoframe_pack(&frame, buf + 1, sizeof(buf) - 1);
 	if (ret < 0) {
 		DRM_ERROR("failed to pack audio infoframe: %d\n", ret);
-- 
1.7.9.5

